rectools/models/nn/transformers/similarity.py

```diff
@@ -0,0 +1,109 @@
+import typing as tp
+
+import numpy as np
+import torch
+from scipy import sparse
+
+from rectools.models.base import InternalRecoTriplet
+from rectools.models.rank import Distance, TorchRanker
+from rectools.types import InternalIdsArray
+
+
+class SimilarityModuleBase(torch.nn.Module):
+    """Similarity module base."""
+
+    def _get_full_catalog_logits(self, session_embs: torch.Tensor, item_embs: torch.Tensor) -> torch.Tensor:
+        raise NotImplementedError()
+
+    def _get_pos_neg_logits(
+        self, session_embs: torch.Tensor, item_embs: torch.Tensor, item_ids: torch.Tensor
+    ) -> torch.Tensor:
+        raise NotImplementedError()
+
+    def forward(
+        self, session_embs: torch.Tensor, item_embs: torch.Tensor, item_ids: tp.Optional[torch.Tensor] = None
+    ) -> torch.Tensor:
+        """Forward pass to get logits."""
+        raise NotImplementedError()
+
+    def _recommend_u2i(
+        self,
+        user_embs: torch.Tensor,
+        item_embs: torch.Tensor,
+        user_ids: InternalIdsArray,
+        k: int,
+        sorted_item_ids_to_recommend: InternalIdsArray,
+        ui_csr_for_filter: tp.Optional[sparse.csr_matrix],
+    ) -> InternalRecoTriplet:
+        """Recommend to users."""
+        raise NotImplementedError()
+
+
+class DistanceSimilarityModule(SimilarityModuleBase):
+    """Distandce similarity module."""
+
+    dist_available: tp.List[str] = ["dot", "cosine"]
```

Comment:
```suggestion
    dist_available: tp.List[Distance] = [Distance.DOT, Distance.COSINE]
```
(After refactoring Distance)